lib/dal/src/func/backend.rs

```diff
@@ -88,6 +87,7 @@ pub enum FuncBackendKind {
     JsAttribute,
     JsAuthentication,
     Json,
+    // NOTE(nick): this has been deprecated. Not adding serde deprecated tag in case it affects the type.
     JsReconciliation,
```

Comment:
We should also probably remove the `#[remain::sorted]`, since we actually only want to add to the *end* of this enum (and the one below), and maybe add a note that changing this enum will change a type that goes into the content store and so should be done with care